pkg/registry/pod/strategy.go

```diff
@@ -291,12 +291,12 @@ func LogLocation(
 			return nil, nil, errors.NewBadRequest(fmt.Sprintf("container %s is not valid for pod %s", container, name))
 		}
 	}
-	nodeHost := pod.Spec.NodeName
-	if len(nodeHost) == 0 {
+	nodeName := pod.Spec.NodeName
+	if len(nodeName) == 0 {
 		// If pod has not been assigned a host, return an empty location
 		return nil, nil, nil
 	}
-	nodeScheme, nodePort, nodeTransport, err := connInfo.GetConnectionInfo(ctx, nodeHost)
+	connectionInfo, err := connInfo.GetConnectionInfo(ctx, nodeName)
```

Comment:
Aha!  So the thing is that here connInfo is a REST object, not a "raw" HTTPKubeletClient object.  Because the one will convert the the nodeName and the other will not.  I'm going to verify that; but I propose changing REST to implement a different interface, given the behaviours are different (convert node name vs not).